### PR TITLE
Backport 22088 to master

### PR DESCRIPTION
**Pull request by moidx**

### CODE REVIEW

**Review by timothytrippel — APPROVED** [review]

**Comment by RyanTorok** [comment]
The merge conflict appears to have been caused by a few SCA test build rules that were previously only on the sival branch.

I found the merge conflict can be avoided if we cherry-pick #21642 and then #21733. However, this still fails to build on master with the following error: ` target 'silicon_owner_proda_rom_ext' not declared in package 'hw/top_earlgrey'`.

**Review by milesdai — APPROVED** [review]

File: sw/device/tests/crypto/cryptotest/firmware/BUILD [hunk 1]
```diff
@@ -250,42 +249,50 @@ cc_library(
...
+        srcs = [":firmware.c"],
+        exec_env = [
+            "//hw/top_earlgrey:{}".format(exec_env),
+        ],
+        deps = FIRMWARE_DEPS,
+    )
+    for exec_env in [
+        "fpga_cw310_test_rom",
+        "silicon_owner_sival_rom_ext",
+        "silicon_owner_proda_rom_ext",
```
Comment (milesdai) [inline sw/device/tests/crypto/cryptotest/firmware/BUILD:292]: Should we remove this for the master branch?

Comment (RyanTorok) [reply sw/device/tests/crypto/cryptotest/firmware/BUILD:292]: I confirmed the firmware builds successfully with the `proda` entry removed.

Comment (moidx) [reply sw/device/tests/crypto/cryptotest/firmware/BUILD]: Yep. This should be removed. Done.

**Comment by moidx** [comment]
Rebase with the cherry picks for https://github.com/lowRISC/opentitan/pull/21642 and  https://github.com/lowRISC/opentitan/pull/21733 to align the commit history.

**Review by RyanTorok — APPROVED** [review]

**Pull request merged**